shape/src/commonMain/kotlin/Arc.kt

```diff
@@ -0,0 +1,291 @@
+package com.juul.krayon.shape
+
+import com.juul.krayon.kanvas.Path
+import com.juul.krayon.kanvas.PathBuilder
+import kotlin.math.abs
+import kotlin.math.acos
+import kotlin.math.asin
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.sin
+import kotlin.math.sqrt
+
+private const val PI = kotlin.math.PI.toFloat()
+private const val HALF_PI = PI / 2
+private const val TAU = 2 * PI
+
+private const val EPSILON = 1e-6f // Number that's close to zero
+
+public fun arc(
+    outerRadius: Float,
+    innerRadius: Float = 0f,
+): Arc = Arc(outerRadius, innerRadius)
+
+public class Arc internal constructor(
+    outerRadius: Float,
+    innerRadius: Float,
+) {
+    public var outerRadius: Float = outerRadius
+        private set
+    public var innerRadius: Float = innerRadius
+        private set
+    public var cornerRadius: Float = 0f
+        private set
+    public var padRadius: (arc: Arc, startAngle: Float, endAngle: Float, padAngle: Float) -> Float = ::defaultPadRadius
+        private set
+
+    public fun outerRadius(value: Float): Arc = apply { outerRadius = value }
+    public fun innerRadius(value: Float): Arc = apply { innerRadius = value }
+    public fun cornerRadius(value: Float): Arc = apply { cornerRadius = value }
+    public fun padRadius(value: (Arc, Float, Float, Float) -> Float): Arc = apply { padRadius = value }
+
+    public operator fun invoke(slice: Slice<*>): Path = this(slice.startAngle, slice.endAngle, slice.padAngle)
+
+    public operator fun invoke(startAngle: Float, endAngle: Float, padAngle: Float): Path = Path {
+        // Implementation stolen from D3. https://github.com/d3/d3-shape/blob/main/src/arc.js#L87
```

Comment:
nit

Sounds friendlier as: 🙃 

```suggestion
        // Implementation "borrowed" from D3. https://github.com/d3/d3-shape/blob/main/src/arc.js#L87
```